spec/lib/time_util_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe TimeUtil do
+  let!(:client_occurred_at) { '2020-10-07T23:30:06+00:00' }
```

Comment:
Recommend rewriting all of the values starting with seconds since epoch to make it easier to see the right answer

```ruby
describe `.device_adjust` do
  allow(Time).to receive(:now).and_return(Time.at(38))
  subject(:server_occurred_at) { 
    described_class.device_adjust(client_clock_occurred_at: Time.at(42).iso8601, 
                                  client_clock_sent_at: Time.at(45).iso8601)
  }
  expect(server_occurred_at).to eq Time.at(35)
end
```

Reply:
ok...